src/lib/registration/registration_ui.rb

```diff
@@ -294,13 +294,16 @@ def register_selected_addons(selected_addons, known_reg_codes)
       registration_order = selected_addons.clone
 
       product_succeed = registration_order.map do |product|
-        ConnectHelpers.catch_registration_errors(
+        registered = ConnectHelpers.catch_registration_errors(
           message_prefix: "#{product.label}\n") do
-          register_selected_addon(product, known_reg_codes[product.identifier])
-        end
+            register_selected_addon(product, known_reg_codes[product.identifier])
+          end
 
         # remove from selected after successful registration
-        selected_addons.reject! { |selected| selected.identifier == product.identifier }
+        if registered
+          selected_addons.reject! { |selected| selected.identifier == product.identifier }
+        end
+        registered
```

Comment:
Exactly, before, it ever deleted the product from the selected addons, so now it register all with good codes and keep not register ones to be filled if you want to.